util/lockwaiter/lockwaiter.go

```diff
@@ -145,3 +159,19 @@ func (lw *Manager) CleanUp(w *Waiter) {
 		q.removeWaiter(w)
 	}
 }
+
+// WakeUpDetection wakes up waiters waiting for deadlock detection by sending
+// rpc request to remote leader
+func (lw *Manager) WakeUpDetection(resp *deadlock.DeadlockResponse) {
```

Comment:
I think `WakeUpForDeadlock` is better, we wake up the waiter because we meet a deadlock error.